flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/DefaultCatalogTable.java

```diff
@@ -40,15 +40,27 @@ public class DefaultCatalogTable implements CatalogTable {
     private final List<String> partitionKeys;
     private final Map<String, String> options;
 
+    private final @Nullable Long snapshot;
```

Comment:
@hackergin please also update `CatalogPropertiesUtil.serializeCatalogTable(this)` otherwise we break the contract that `ResolvedCatalogTable` is fully serializable.

Reply:
CC @luoyuxia

Reply:
Thanks for reminder. We will fix update it in FLINK-32569

Reply:
I submitted the fix pr.  https://github.com/apache/flink/pull/22977